projects/packages/forms/tests/php/contact-form/Contact_Form_Test.php

```diff
@@ -903,12 +880,87 @@ public function test_make_sure_checkbox_multiple_field_renders_as_expected() {
 
 		$expected_attributes = array_merge( $attributes, array( 'input_type' => 'checkbox' ) );
 		$this->assertValidFieldMultiField( $this->render_field( $attributes ), $expected_attributes );
+
+		// Test with block attributes.
+		$block = array(
+			'blockName'   => 'jetpack/field-checkbox-multiple',
+			'attrs'       => array(
+				'required'             => false,
+				'shareFieldAttributes' => false,
+				'className'            => 'is-style-list',
+			),
+			'innerBlocks' => array(
+				array(
+					'blockName' => 'jetpack/label',
+					'attrs'     => array(
+						'label'        => 'Choose several options',
+						'defaultLabel' => 'Add label…',
+						'textColor'    => 'swamp-green',
+						'style'        => array(
+							'elements' => array(
+								'link' => array( 'color' => array( 'text' => 'var:preset|color|accent-3' ) ),
+							),
+						),
+					),
+				),
+				array(
+					'blockName'   => 'jetpack/options',
+					'attrs'       => array(
+						'type'  => 'checkbox',
+						'style' => array(
+							'spacing' => array(
+								'blockGap' => 'var:preset|spacing|40',
+							),
+						),
+					),
+					'innerBlocks' => array(
+						array(
+							'blockName' => 'jetpack/option',
+							'attrs'     => array(
+								'label' => 'option 1',
+								'style' => array(
+									'color'      => array( 'text' => 'caramel' ),
+									'elements'   => array(
+										'link' => array( 'color' => array( 'text' => 'caramel' ) ),
+									),
+									'typography' => array(
+										'fontSize' => '24px',
+									),
+								),
+							),
+						),
+						array(
+							'blockName' => 'jetpack/option',
+							'attrs'     => array(
+								'label' => 'option 2',
+								'style' => array(
+									'color'      => array( 'text' => 'gummy' ),
+									'elements'   => array(
+										'link' => array( 'color' => array( 'text' => 'gummy' ) ),
+									),
+									'typography' => array(
+										'fontSize' => '24px',
+									),
+								),
+							),
+						),
+					),
+				),
+			),
+		);
+
+		// Convert block attributes to shortcode attributes.
+		$attributes_with_block_attributes = Contact_Form_Plugin::block_attributes_to_shortcode_attributes( $attributes, 'checkbox-multiple', new \WP_Block( $block ) );
```

Comment:
Is there a better way to convert block attributes to attributes the old form field expects?